src/common/tests/umem_test.c

```diff
@@ -0,0 +1,252 @@
+
+#define POOL_SIZE ((1024 * 1024  * 1024ULL))
+
+struct test_arg {
+	struct utest_context	*ta_utx;
+	uint64_t		*ta_root;
+	char			*ta_pool_name;
+};
+
+int
+teardown_vmem(void **state)
+{
+	struct test_arg *arg = *state;
+	int		 rc;
+
+	if (arg == NULL) {
+		print_message("state not set, likely due to group-setup"
+			      " issue\n");
+		return 0;
+	}
+
+	rc = utest_utx_destroy(arg->ta_utx);
+
+	return rc;
+}
+int
+setup_vmem(void **state)
+{
+	struct test_arg		*arg = *state;
+	int			 rc = 0;
+
+	rc = utest_vmem_create(sizeof(*arg->ta_root), &arg->ta_utx);
+	if (rc != 0) {
+		perror("Could not create vmem context");
+		rc = 1;
+		goto failed;
+	}
+
+	arg->ta_root = utest_utx2root(arg->ta_utx);
+
+	return 0;
+failed:
+	return rc;
+}
+
+
+int
+teardown_pmem(void **state)
+{
+	struct test_arg *arg = *state;
+	int		 rc;
+
+	if (arg == NULL) {
+		print_message("state not set, likely due to group-setup"
+			      " issue\n");
+		return 0;
+	}
+
+	rc = utest_utx_destroy(arg->ta_utx);
+
+	D_FREE(arg->ta_pool_name);
+
+	return rc;
+}
+int
+setup_pmem(void **state)
+{
+	struct test_arg		*arg = *state;
+	static int		 tnum;
+	int			 rc = 0;
+
+	D_ASPRINTF(arg->ta_pool_name, "/mnt/daos/umem-test-%d", tnum++);
+	if (arg->ta_pool_name == NULL) {
+		print_message("Failed to allocate test struct\n");
+		return 1;
+	}
+
+	rc = utest_pmem_create(arg->ta_pool_name, POOL_SIZE,
+			       sizeof(*arg->ta_root), &arg->ta_utx);
+	if (rc != 0) {
+		perror("Could not create pmem context");
+		rc = 1;
+		goto failed;
+	}
+
+	arg->ta_root = utest_utx2root(arg->ta_utx);
+
+	return 0;
+failed:
+	D_FREE(arg->ta_pool_name);
+	return rc;
+}
+
+static int
+global_setup(void **state)
+{
+	struct test_arg	*arg;
+
+	D_ALLOC_PTR(arg);
+	if (arg == NULL) {
+		print_message("Failed to allocate test struct\n");
+		return 1;
+	}
+
+	*state = arg;
+
+	return 0;
+}
+
+static int
+global_teardown(void **state)
+{
+	struct test_arg	*arg = *state;
+
+	D_FREE(arg);
+
+	return 0;
+}
+
+static void
+test_invalid_flags(void **state)
+{
+	struct test_arg		*arg = *state;
+	umem_off_t		 umoff = 0;
+	int			 i;
+
+	/* 0 is not considered NULL for an offset */
+	assert_false(UMOFF_IS_NULL(umoff));
+
+	assert_int_equal(umem_off_get_invalid_flags(umoff), 0);
+
+	for (i = 0; i < UMOFF_NUM_FLAGS; i++) {
+		umem_off_set_invalid(&umoff, 1 << i);
+		assert_int_equal(umem_off_get_invalid_flags(umoff), 1 << i);
+	}
+
+	umoff = UMOFF_NULL;
+	assert_int_equal(umem_off_get_invalid_flags(umoff), 0);
+
+	assert_int_equal(utest_alloc_off(arg->ta_utx, &umoff, 4, NULL, NULL),
+			 0);
+	assert_int_equal(umem_off_get_invalid_flags(umoff), 0);
+
+	assert_int_equal(utest_free_off(arg->ta_utx, umoff), 0);
+}
+
+static void
+test_alloc(void **state)
+{
+	struct test_arg		*arg = *state;
+	struct umem_instance	*umm = utest_utx2umm(arg->ta_utx);
+	int			*value1;
+	int			*value2;
+	umem_off_t		 umoff = 0;
+	umem_id_t		 mmid;
+	int			 rc;
+
+	rc = utest_tx_begin(arg->ta_utx);
+	assert_int_equal(rc, 0);
+	if (rc != 0)
```

Comment:
I believe assert_int_equal stops execution of the test so this might be redundant.